Library/Homebrew/dev-cmd/pr-pull.rb

```diff
@@ -155,8 +155,24 @@ def run
                 end
 
                 ohai "Downloading bottles for workflow: #{workflow}"
-                url = GitHub.get_artifact_url(workflow_run)
-                GitHub.download_artifact(url, pr)
+
+                # Before migration to artifact actions v4, failed bottles were
+                # uploaded with artifact names matching `bottles-*`. Therefore,
+                # we should prioritise the `bottles` artifact, if it exists, to
+                # avoid picking up the failed bottles.
+                urls = if tap.core_tap?
+                  old_workflow_run = workflow_run.dup
+                  old_workflow_run[-1] = "bottles"
+                  begin
+                    GitHub.get_artifact_urls(old_workflow_run)
+                  rescue
+                    nil
+                  end
+                end
+                urls ||= GitHub.get_artifact_urls(workflow_run)
```

Comment:
Have changed the default pattern to `bottles-*` to avoid matching the old name which could cause problems.

Instead, added an extra logic here to always download `bottles` first, and only attempt to download `bottles-*` if `bottles` does not exist.

This also means `pr-pull` should never be run on PRs running the old workflow with no `bottles`. But we don't do that anyway.